retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaCallFactory.java

```diff
@@ -79,8 +97,9 @@ public Call newCall(Request request) {
             this.request = request;
         }
 
-        private static HttpResponse doCall(HttpClient httpClient, Request request) {
-            URL url = request.url().url();
+        private static HttpResponse doCall(ArmeriaCallFactory callFactory, Request request) {
+            URI url = request.url().uri();
+            HttpClient httpClient = callFactory.getHttpClient(url.getAuthority());
```

Comment:
Ah found the documentation about base-url resolution (http://square.github.io/retrofit/2.x/retrofit/retrofit2/Retrofit.Builder.html#baseUrl-okhttp3.HttpUrl-) - how about creating a derived client here if the authority differs from the base HttpClient?